src/o365/spo/commands/tenant/tenant-settings-set.ts

```diff
@@ -0,0 +1,861 @@
+            authorization: `Bearer ${auth.service.accessToken}`,
+            'X-RequestDigest': formDigestValue
+          }),
+          body: `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><ObjectPath Id="4" ObjectPathId="3" /><Query Id="5" ObjectPathId="3"><Query SelectAllProperties="true"><Properties /></Query></Query></Actions><ObjectPaths><Constructor Id="3" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}" /></ObjectPaths></Request>`
+        };
+
+        if (this.debug) {
+          cmd.log('Retrieved ContextInfo...');
+          cmd.log(contextResponse);
+          cmd.log('');
+        }
+
+        return request.post(requestOptions);
+      })
+      .then((res: any): request.RequestPromise | Promise<any> => {
+
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(res);
+          cmd.log('');
+        }
+
+        const json: ClientSvcResponse = JSON.parse(res);
+        const response: ClientSvcResponseContents = json[0];
+        if (response.ErrorInfo) {
+          const error: string = response.ErrorInfo.ErrorMessage;
+          return new Promise<any>((resolve, reject) => { reject(error) });
+        }
+
+        // map the args.options to XML Properties
+        let propsXml: string = '';
+        let id = 42; // geek's humor
+        for (let optionKey of Object.keys(args.options)) {
+
+          if (this.isExcludedOption(optionKey)) {
+            continue;
+          }
+
+          let optionValue = args.options[optionKey];
+          if (this.getAllEnumOptions().indexOf(optionKey) > -1) {
+            // map enum values to int
+            optionValue = this.mapEnumToInt(optionKey, args.options[optionKey]);
+          }
+
+          if (['AllowedDomainListForSyncClient', 'DisabledWebPartIds'].indexOf(optionKey) > -1) {
+
+            // the XML has to be represented as array of guids
+            let valuesXml = '';
+            optionValue.split(',').forEach((value: string) => {
+              valuesXml += `<Object Type="Guid">{${value}}</Object>`;
+            })
+            propsXml += `<SetProperty Id="${id}" ObjectPathId="7" Name="${optionKey}"><Parameter Type="Array">${valuesXml}</Parameter></SetProperty><Method Name="Update" Id="${id + 1}" ObjectPathId="7" />`;
+
+            id += 2;
+
+          } else if (['ExcludedFileExtensionsForSyncClient'].indexOf(optionKey) > -1) {
+
+            // the XML has to be represented as array of strings
+            let valuesXml = '';
+            optionValue.split(',').forEach((value: string) => {
+              valuesXml += `<Object Type="String">${value}</Object>`;
+            })
+            propsXml += `<SetProperty Id="${id}" ObjectPathId="7" Name="${optionKey}"><Parameter Type="Array">${valuesXml}</Parameter></SetProperty><Method Name="Update" Id="${id + 1}" ObjectPathId="7" />`;
```

Comment:
Simplify increasing the value of `id`, by changing all instances to `id++`, which will cause the value to be increased after each time the variable was used in an XML string. This way you don't need to keep track by how much the variable value should be increased.

Reply:
Nice, that is smart!